dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPIImpl.java

```diff
@@ -147,9 +148,19 @@ private void checkHostPermission(final User user, boolean respectAnonPerms, fina
 
 	    if (hostId != null && user.isBackendUser()) {
 	        return Optional.ofNullable(find(hostId, user, respectAnonPerms));
-        } else if (request.getParameter(Host.HOST_VELOCITY_VAR_NAME) != null) {
-            final String hostName = request.getParameter(Host.HOST_VELOCITY_VAR_NAME);
-            return this.resolveHostNameWithoutDefault(hostName, user, respectAnonPerms);
+        } else if (UtilMethods.isSet(request.getParameter(Host.HOST_VELOCITY_VAR_NAME))
+            || UtilMethods.isSet(request.getAttribute(Host.HOST_VELOCITY_VAR_NAME))) {
+
+	        final String hostIdOrName = UtilMethods.isSet(

```

Comment:
I will cache `request.getParameter(Host.HOST_VELOCITY_VAR_NAME)` in a variable to avoid search twice using `request.getParameter()`.

Reply:
Will do